### PR TITLE
Fix route53 support

### LINKED ISSUE #715: Let's Encrypt Addon does not work with AWS Route53
The log produced the following error:
```
usage: 
  certbot [SUBCOMMAND] [options] [-d DOMAIN] [-d DOMAIN] ...
Certbot can obtain and install HTTPS/TLS/SSL certificates.  By default,
it will attempt to use a webserver both for obtaining and installing the
certificate. 
certbot: error: unrecognized arguments: --dns-route53-credentials /data/dnsapikey
```
I will be making a pull request shortly.

### PR DESCRIPTION
**Pull request by Zapfmeister**
closes #715

### CODE REVIEW

**Comment by Zapfmeister** [comment]
closes #715

**Comment by kre8or69** [comment]
Would love this to be merged in! Just started using Hassio and already use Route53! haha

File: letsencrypt/run.sh [hunk 1]
```diff
@@ -13,38 +13,45 @@ WORK_DIR=/data/workdir
...
-  "dns_rfc2136_algorithm = $(bashio::config 'dns.rfc2136_algorithm')\n" \
-  "aws_access_key_id = $(bashio::config 'dns.aws_access_key_id')\n" \
-  "aws_secret_access_key = $(bashio::config 'dns.aws_secret_access_key')\n" \
-  "dns_sakuracloud_api_token = $(bashio::config 'dns.sakuracloud_api_token')\n" \
-  "dns_sakuracloud_api_secret = $(bashio::config 'dns.sakuracloud_api_secret')" > /data/dnsapikey
-chmod 600 /data/dnsapikey
+if bashio::config 'dns.provider' == "dns-route53"; then
+    touch ~/.aws/config
+    echo -e "[default]\n" \
+        "aws_access_key_id = $(bashio::config 'dns.aws_access_key_id')\n" \
```
Comment (frenck) [inline letsencrypt/run.sh]: So do we require `dns.aws_access_key_id` & `dns.aws_secret_access_key` to be set when the `dns.provider` = `dns-route53`?

If so, we need a check for this.

File: letsencrypt/run.sh [hunk 2]
```diff
@@ -13,38 +13,45 @@ WORK_DIR=/data/workdir
...
-  "dns_rfc2136_port = $(bashio::config 'dns.rfc2136_port')\n" \
-  "dns_rfc2136_name = $(bashio::config 'dns.rfc2136_name')\n" \
-  "dns_rfc2136_secret = $(bashio::config 'dns.rfc2136_secret')\n" \
-  "dns_rfc2136_algorithm = $(bashio::config 'dns.rfc2136_algorithm')\n" \
-  "aws_access_key_id = $(bashio::config 'dns.aws_access_key_id')\n" \
-  "aws_secret_access_key = $(bashio::config 'dns.aws_secret_access_key')\n" \
-  "dns_sakuracloud_api_token = $(bashio::config 'dns.sakuracloud_api_token')\n" \
-  "dns_sakuracloud_api_secret = $(bashio::config 'dns.sakuracloud_api_secret')" > /data/dnsapikey
-chmod 600 /data/dnsapikey
+if bashio::config 'dns.provider' == "dns-route53"; then
```
Comment (frenck) [inline letsencrypt/run.sh]: This logic now adds only these 2 settings, while previous behavior added all configuration of each provider to the configuration.

This this correct behavior? Does this mean only 1 provider can be active at the same time? If that is the case, the logic below (the else) should be adjusted as well to preview major loading of configs (and thus unnecessary calls). If not, well than the else should be removed I guess.

File: letsencrypt/run.sh [hunk 3]
```diff
@@ -13,38 +13,45 @@ WORK_DIR=/data/workdir
 mkdir -p "$WORK_DIR"
 mkdir -p "$CERT_DIR"
 mkdir -p "/ssl"
+mkdir -p ~/.aws/
```
Comment (frenck) [inline letsencrypt/run.sh]: We can combine those `mkdir -p`'s into a single `mkdir` call.

File: letsencrypt/run.sh [hunk 4]
```diff
@@ -54,8 +61,12 @@ if [ ! -d "$CERT_DIR/live" ]; then
     done
 
     echo "$DOMAINS" > /data/domains.gen
-    if [ "$CHALLENGE" == "dns" ]; then
-        certbot certonly --non-interactive --config-dir "$CERT_DIR" --work-dir "$WORK_DIR" "--$DNS_PROVIDER" "--${DNS_PROVIDER}-credentials" "/data/dnsapikey" --email "$EMAIL" --agree-tos --config-dir "$CERT_DIR" --work-dir "$WORK_DIR" --preferred-challenges "$CHALLENGE" "${DOMAIN_ARR[@]}"
+    if bashio::config 'challenge' == "dns"; then
```
Comment (frenck) [inline letsencrypt/run.sh]: This comparison would not work. Have you tested the add-on?

File: letsencrypt/run.sh [hunk 5]
```diff
@@ -54,8 +61,12 @@ if [ ! -d "$CERT_DIR/live" ]; then
...
+        if bashio::config 'dns_provider' == "dns-route53"; then
```
Comment (frenck) [inline letsencrypt/run.sh]: same

File: letsencrypt/run.sh [hunk 6]
```diff
@@ -54,8 +61,12 @@ if [ ! -d "$CERT_DIR/live" ]; then
...
+            certbot certonly --non-interactive --config-dir "$CERT_DIR" --work-dir "$WORK_DIR" "--$DNS_PROVIDER" --email "$EMAIL" --agree-tos --config-dir "$CERT_DIR" --work-dir "$WORK_DIR" --preferred-challenges "$CHALLENGE" "${DOMAIN_ARR[@]}"
```
Comment (frenck) [inline letsencrypt/run.sh]: Add some linebreaks to make this more readable.

File: letsencrypt/run.sh [hunk 7]
```diff
@@ -54,8 +61,12 @@ if [ ! -d "$CERT_DIR/live" ]; then
...
+        else
+            certbot certonly --non-interactive --config-dir "$CERT_DIR" --work-dir "$WORK_DIR" "--$DNS_PROVIDER" "--${DNS_PROVIDER}-credentials" "/data/dnsapikey" --email "$EMAIL" --agree-tos --config-dir "$CERT_DIR" --work-dir "$WORK_DIR" --preferred-challenges "$CHALLENGE" "${DOMAIN_ARR[@]}"
```
Comment (frenck) [inline letsencrypt/run.sh]: Add some linebreaks to make this more readable.

File: letsencrypt/run.sh (hunk 1 above)
Comment (Zapfmeister) [reply letsencrypt/run.sh]: Yes, those are the required variables for route53

File: letsencrypt/run.sh (hunk 2 above)
Comment (Zapfmeister) [reply letsencrypt/run.sh]: Your right. Thanks for the hint. It is now fixed

File: letsencrypt/run.sh (hunk 4 above)
Comment (Zapfmeister) [reply letsencrypt/run.sh]: Yes, but it slipped my eyes, as it was always true and i just tested against dns.
Fixed.

File: letsencrypt/run.sh (hunk 6 above)
Comment (Zapfmeister) [reply letsencrypt/run.sh]: Fixed

File: letsencrypt/run.sh (hunk 7 above)
Comment (Zapfmeister) [reply letsencrypt/run.sh]: Fixed

File: letsencrypt/run.sh (hunk 5 above)
Comment (Zapfmeister) [reply letsencrypt/run.sh]: Fixed

**Comment by Zapfmeister** [comment]
@frenck could you please have a look? It should be fine now for merging

**Comment by tenacioustechie** [comment]
Any chance this can be updated and merged @Zapfmeister  and @frenck ?
Just hit this same issue on route53, let me know if there is anything I can do.

**Comment by pvizeli** [comment]
Is fixed

**Comment by BarrettLowe** [comment]
I'm still having this problem. Is the fix released yet?

**Comment by imduffy15** [comment]
This does not appear to be fixed, https://github.com/home-assistant/hassio-addons/commit/50c295eee71cda83c42820e7e2b8ffb56df05aa1#diff-4d929de130b59358dd27cea01a3cff10 introduces a workaround for it, however, the --dns-route53-credentials argument is still passed which is not valid.

I've copied the code and made my own repo to work around it for now, fix is available here https://github.com/imduffy15/hassio-addons/commit/2bade84490ae8cb9469c1b25e9a937789e787d23 or if you want it as a hass repo (don't rely on it long term) https://github.com/imduffy15/hassio-addons

**Comment by borger** [comment]
It doesn't appear to be fixed yet... :(

**Pull request closed without merging**